src/recipe/lua.rs

```diff
@@ -0,0 +1,276 @@
+use crate::config::{Contract, TemplateType};
+use crate::generator::{CreateContract, TEMPLATES};
+use crate::project_context::{BuildConfig, BuildEnv, Context, CONTRACTS_DIR};
+use crate::recipe::Recipe;
+use crate::signal::Signal;
+use crate::util::cli;
+use crate::util::git;
+use anyhow::{anyhow, Result};
+use std::fs;
+use std::io::Write;
+use std::marker::PhantomData;
+use std::path::PathBuf;
+use tera;
+
+// Files
+
+const MAKEFILE: &str = "Makefile";
+
+// Dirs
+
+const LUA_DIR_PREFIX: &str = "lua";
+const LUA_TEMPLATE_DIR_PREFIX: &str = "lua";
+const DEPS_DIR_PREFIX: &str = "deps";
+const SRC_DIR_PREFIX: &str = "src";
+const DEBUG_DIR: &str = "build/debug";
+const RELEASE_DIR: &str = "build/release";
+
+// Deps
+
+const CKB_LUA_URL: &str = "https://github.com/nervosnetwork/ckb-lua.git";
+const CKB_LUA_COMMIT: &str = "ffc147e6a091a7a90b7dbe28d0a140def336bc7f";
+const CKB_LUA_NAME: &str = "ckb-lua";
+
+pub trait LuaRecipe {
+    fn bin_name(name: &str) -> String;
+    fn src_template() -> &'static str;
+    fn build_template() -> &'static str;
+}
+pub struct LuaStandalone;
+
+impl LuaRecipe for LuaStandalone {
+    fn bin_name(name: &str) -> String {
+        name.to_string()
+    }
+
+    fn src_template() -> &'static str {
+        "standalone/contract/example.lua"
+    }
+
+    fn build_template() -> &'static str {
+        "standalone/contract/BUILD"
```

Comment:
why build is in uppercase

Reply:
It is initially copied that from the c shared library template. I can change that if it is wanted.